rpc/src/v1/extractors.rs

```diff
@@ -36,9 +36,9 @@ pub struct RpcExtractor;
 impl HttpMetaExtractor for RpcExtractor {
 	type Metadata = Metadata;
 
-	fn read_metadata(&self, _origin: Option<String>, user_agent: Option<String>, _dapps_origin: Option<String>) -> Metadata {
+	fn read_metadata(&self, origin: Option<String>, user_agent: Option<String>) -> Metadata {
 		Metadata {
-			origin: match user_agent {
+			origin: match user_agent.or(origin) {
```

Comment:
IMHO should be other way around, `Origin` provides more info than `UserAgent`. Maybe best just do:
```
let service = format!("{} / {}", origin.unwrap_or("unknown origin"), user_agent.unwrap_or("unknown agent"));
```
?